core/controllers/topic_editor.py

```diff
@@ -151,6 +152,23 @@ def get(self, topic_id: str) -> None:
 
         updated_canonical_story_summary_dicts = []
         for summary in canonical_story_summary_dicts:
+            story = story_fetchers.get_story_by_id(summary['id'])
+            nodes = story.story_contents.nodes
+            total_chapters_count = len(nodes)
+            published_chapters_count = len(
+                [node for node in nodes if
+                node.status == 'Published'])
+            upcoming_chapters_count = len(
+                [node for node in nodes if
+                node.planned_publication_date is not None and (
+                node.planned_publication_date - datetime.datetime.today()).
+                days < 14 and node.planned_publication_date >
+                datetime.datetime.today() and node.status != 'Published'])
+            overdue_chapters_count = len(
```

Comment:
Instead of doing 3 list comprehensions, perhaps do a single for loop through the chapters and increment the counts (using if/else logic)? That would be more efficient by a factor of 3, and also probably easier to read since some of the long conditions are repeated.

Reply:
Done